python/tvm/script/builder/tir/op.py

```diff
@@ -30,16 +30,25 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+def dtype_forward(func):
+    def forwarded(*args, **kwargs):
+        if "dtype" in kwargs:
+            args = (kwargs.get("dtype"),) + args
+            kwargs.pop("dtype")
```

Comment:
i guess `pop` has return values
 
```suggestion
            args = (kwargs.pop("dtype"),) + args
```